cirq-core/cirq/ops/gate_operation.py

```diff
@@ -258,6 +258,18 @@ def _measurement_key_names_(self) -> Optional[Iterable[str]]:
             return getter()
         return NotImplemented
 
+    def _measurement_key_obj_(self) -> Optional[value.MeasurementKey]:
+        getter = getattr(self.gate, '_measurement_key_obj_', None)
+        if getter is not None:
+            return getter()
+        return NotImplemented
+
+    def _measurement_key_objs_(self) -> Optional[Iterable[value.MeasurementKey]]:
```

Comment:
Should we be consistent with the signature? Some are sets, some are iterable, some are unmarked.

Reply:
Moved all I could find to `AbstractSet` and explicit type annotations. Let me know if you find instances that I might have missed. 

Initially I was concerned that changing `Iterable` to `AbstractSet` might be type-breaking (which isn't *that* serious given python) for external usages of these method. But since these protocols and magic methods were just recently added, they haven't been released externally. The internal usage of the now deprecated `_measurement_keys_` method explicitly wraps it in a `set`, so all external code using `_measurement_key_` should work as well.